File: Algorithms/Unionoflists/algorithm3.cpp
#include <bits/stdc++.h>
#include <time.h>
using namespace std;
int main()
{
  srand(time(0));
  int l1=rand()%10;
  int l2=rand()%10;
  int A[l1],B[l2];
  for(int i=0;i<l1;i++)
    A[i]=rand()%10-rand()%10;
  for(int i=0;i<l2;i++)
    B[i]=rand()%10-rand()%10;
  sort(A,A+l1);
  sort(B,B+l2);
  for(int i=0;i<l1;i++)
    cout<<A[i]<<',';
  cout<<"\b \n";
  for(int i=0;i<l2;i++)
    cout<<B[i]<<',';
  cout<<"\b \n";
  int temp=0;
//Algorithm 3: sort both arrays and iterate through both the lists
  int len=min(l1,l2);
  int l=0,m=0;
  for(int i=0;i<len;i++)
  {
    if(A[l]<B[m])
      l++;
    else if(A[l]>B[m])
      m++;
    else
    {
      l++;
      m++;
      temp++;
    }
  }
  cout<<"Number of common elements=" <<temp;
}

File: Algorithms/Unionoflists/algorithm3.py
import random
A=random.sample(range(-10,10),random.randint(0,10))
A.sort()
B=random.sample(range(-10,10),random.randint(0,10))
B.sort()
print(A)
print(B)
temp=0
#Algorithm 3: sort both arrays and iterate through both the lists
l=0
m=0
unionAB=[]
while l<len(A) and m<len(B):
    if A[l]<B[m]:
        l=l+1
    elif A[l]>B[m]:
        m=m+1
    else:
        temp=temp+1
        unionAB.append(A[l])
        l=l+1
        m=m+1
print(unionAB)
